### PR TITLE
Allow members to remove themselves from projects

### PR DESCRIPTION
**Pull request by VPashkov**
# Description 📣

## Type ✨

- [ ] Bug fix
- [x] New feature
- [ ] Breaking change
- [ ] Documentation

# Tests 🛠️

Tested manually:
- Users cannot delete other users if they don't have admin role
- Users cannot delete the last admin in the workspace
- Users can delete other users if they have admin role
- Users can delete themselves
- Users are redirected to the home page after deleting themselves

---

- [x] I have read the [contributing guide](https://infisical.com/docs/contributing/overview), agreed and acknowledged the [code of conduct](https://infisical.com/docs/contributing/code-of-conduct). 📝

### CODE REVIEW

**Comment by dangtony98** [comment]
Hi @VPashkov!

This is pretty close! — However, when I in the role of an `admin` tried adding a new member to my project, then removing them, then adding them back, then the member wasn't added back to the project.

Could you check up on this case?

**Comment by VPashkov** [comment]
@dangtony98 I didn't test this scenario. I'll take a look.

**Comment by VPashkov** [comment]
@dangtony98 It looks like this bug is not related to this PR. I can reproduce it without my changes locally. I also can reproduce it at https://app.infisical.com.

File: backend/src/controllers/v1/membershipController.ts
Comment (sheensantoscapadngan) [inline backend/src/controllers/v1/membershipController.ts:95]: hmmm should admin users be able to delete membership of other admin users? @dangtony98

File: frontend/src/components/basic/table/ProjectUsersTable.tsx
Comment (sheensantoscapadngan) [inline frontend/src/components/basic/table/ProjectUsersTable.tsx:68]: 2nd parameter should be `email` as well

**Comment by dangtony98** [comment]
Hi @VPashkov!

Is this PR still in progress? — It looks like another contributor would like to help / take a stab at it.

File: frontend/src/components/basic/table/ProjectUsersTable.tsx
Comment (VPashkov) [reply frontend/src/components/basic/table/ProjectUsersTable.tsx]: Done.

File: frontend/src/components/basic/table/ProjectUsersTable.tsx
Comment (VPashkov) [reply frontend/src/components/basic/table/ProjectUsersTable.tsx]: Done.

**Comment by VPashkov** [comment]
@sheensantoscapadngan I resolved your comments. PTAL

**Review by sheensantoscapadngan — APPROVED** [review]

**Comment by VPashkov** [comment]
Thank you for reviewing @sheensantoscapadngan.
@dangtony98 I think it needs approval from one of the maintainers. Can you please review or assign somebody?

**Comment by dangtony98** [comment]
Hi @VPashkov,

Thanks for this — I'll take a look today :)

**Comment by maidul98** [comment]
Closing as backend has been fully changed from Mongo to Postgres

**Pull request closed without merging**
